platform/mellanox/mlnx-platform-api/sonic_platform/sfp_event.py

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env python
+'''
+listen to the SDK for the SFP change event and return to chassis.
+'''
+
+from __future__ import print_function
+import sys, errno
+import os
+import time
+import select
+from python_sdk_api.sx_api import *
+from sonic_daemon_base.daemon_base import Logger
+
+SYSLOG_IDENTIFIER = "sfp-event"
+
+SDK_SFP_STATE_IN = 0x1
+SDK_SFP_STATE_OUT = 0x2
+STATUS_PLUGIN = '1'
+STATUS_PLUGOUT = '0'
+STATUS_UNKNOWN = '2'
+
+sfp_value_status_dict = {
+        SDK_SFP_STATE_IN:  STATUS_PLUGIN,
+        SDK_SFP_STATE_OUT: STATUS_PLUGOUT,
+}
+
+PMPE_PACKET_SIZE = 2000
+
+logger = Logger(SYSLOG_IDENTIFIER)
+
+class sfp_event:
+    ''' Listen to plugin/plugout cable events '''
+
+    SX_OPEN_RETRIES = 20
+
+    def __init__(self):
+        self.swid = 0
+        self.handle = None
+
+    def initialize(self):
+        # open SDK API handle
```

Comment:
not sure why this is required. is this the way we do for other components when SDK API is to be open?
BTW why 20 times? what makes it be enough? if this is a generic code you ae using on other components i suggest we will have a shared functionality that all use the same.

Also, i suggest to have the comment in 1 line instead of 3.

Reply:
this part is ported from mlnx-sfpd which is in charge of listening to the sfp plugin/out event and committed in [PR 2907](https://github.com/Azure/sonic-buildimage/pull/2907/).
the reason for retrying is that:
1. when the system starts up or after config reload there is no definite order in which sdk and mlnx-sfpd starts. 
2. as a result, it's possible that mlnx-sfpd starts prior to sdk ready.
I don't think to wait until ready is the perfect way to solve racing conditions, but it seems that it's the most simple one and works for mlnx-sfpd. 
I believe this kind of racing condition can exist in the new platform api framework so I ported this code. For the retry times, I suggest having it tested and adjusted when we rewrite the xcvrd daemon based on new api.